lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -796,6 +864,54 @@ namespace {
       return true;
     }
 
+    /// Diagnose an inout argument passed into an async call
+    ///
+    /// \returns true if we diagnosed the entity, \c false otherwise.
+    bool diagnoseInOutArg(const ApplyExpr *call, const InOutArgInfo arg) {
+      // check that the call is actually async
+      if (!isAsyncCall(call))
+        return false;
+
+      Expr *subArg = arg.arg->getSubExpr();
+      // The parameter is a member
+      if (LookupExpr *baseArg = dyn_cast<LookupExpr>(subArg)) {
+        while (LookupExpr *nextLayer = dyn_cast<LookupExpr>(baseArg->getBase()))
+          baseArg = nextLayer;
+        // subArg: the actual property being passed inout
+        // baseArg: the property in the actor who's property is being passed
+        // inout
+
+        auto memberDecl = baseArg->getMember().getDecl();
+        auto isolation = ActorIsolationRestriction::forDeclaration(memberDecl);
+        if (isolation == ActorIsolationRestriction::ActorSelf) {
```

Comment:
I would suggest using an exhaustive switch (no `default` case) instead of this `if` test for actor isolation kind. The benefit of a switch is that if we end up adding more cases in the future, the compiler will point-out all of the places that need updating.